chain/messagepool/messagepool.go

```diff
@@ -794,7 +794,12 @@ func (mp *MessagePool) VerifyMsgSig(m *types.SignedMessage) error {
 		return nil
 	}
 
-	if err := chain.AuthenticateMessage(m, m.Message.From); err != nil {
+	nv, err := mp.getNtwkVersion(mp.curTs.Height())
```

Comment:
I think we need to migrate the message pool as well. Otherwise, we'll pack invalid messages and end up with a bunch of invalid blocks.

Reply:
Hm. I'm not sure if we can really "migrate" the message pool, as we need to handle reverts across the upgrade epoch. Thoughts @arajasek?

Reply:
One solution would be to migrate (well, filter) the mpool when we mint a block.... but then we'd need to detect if/when that's necessary.

Reply:
Wait, no, nvm. Re-orgs don't matter here because an older block will never be _produced_ after the migration epoch. So yes, we can filter the mpool at the migration height.

Reply:
The activity in Hyperspace has decreased drastically. Most tipsets are packing 0-2 messages. I would support a simpler solution, like clearing the mempool and relying on wallets to resubmit messages. Ethereum wallets do this.

Reply:
Yeah, I thought about this, but I think "doing nothing" is the right thing here. IIUC, this will only affect the following messages:

1. Bare sends (method 0) from t410 addresses
1. Contract deployments with empty bytecode.

Eyeballing Filscan, there's ~1 of the first kind of message every hour. I'm not sure how many of the second kind we have, but my guess is virtually none.

Reply:
What would happen in the event of a bad block in a network dominated by a single miner?

Reply:
@raulk That miner would have to manually disable its slash block filter in order to produce a new block. There's also a risk that the miner could be reported for a consensus fault, rendering it unable to mine for a period.

Reply:
Yep. We've discussed it.

- This'll get caught when the miner tries to sync their own block.
- They can then just clear their mpool.